editor/header/saved-state/index.js

```diff
@@ -2,51 +2,72 @@
  * External dependencies
  */
 import { connect } from 'react-redux';
-import classNames from 'classnames';
+import classnames from 'classnames';
 
 /**
  * WordPress dependencies
  */
-import { Dashicon } from 'components';
+import { Dashicon, Button } from 'components';
 
 /**
  * Internal dependencies
  */
 import './style.scss';
-import { isEditedPostNew, isEditedPostDirty } from '../../selectors';
+import { savePost } from '../../actions';
+import {
+	isEditedPostNew,
+	isEditedPostDirty,
+	isSavingPost,
+	getCurrentPost,
+	getPostEdits,
+	getBlocks,
+	getEditedPostStatus,
+} from '../../selectors';
 
-function SavedState( { isNew, isDirty } ) {
-	const classes = classNames( 'editor-saved-state', {
-		'is-new': isNew,
-		'is-dirty': isDirty,
-	} );
+function SavedState( { isNew, isDirty, isSaving, edits, blocks, post, status, onSave } ) {
+	const className = 'editor-saved-state';
 
-	let icon, text;
-	if ( isNew && isDirty ) {
-		icon = 'warning';
-		text = wp.i18n.__( 'New post with changes' );
-	} else if ( isNew ) {
-		icon = 'edit';
-		text = wp.i18n.__( 'New post' );
-	} else if ( isDirty ) {
-		icon = 'warning';
-		text = wp.i18n.__( 'Unsaved changes' );
-	} else {
-		icon = 'saved';
-		text = wp.i18n.__( 'Saved' );
+	if ( isSaving ) {
+		return (
+			<div className={ className }>
+				Saving
+			</div>
+		);
 	}
+	if ( ! isNew && ! isDirty ) {
+		return (
+			<div className={ className }>
+				<Dashicon icon={ 'saved' } />
+				{ wp.i18n.__( 'Saved' ) }
+			</div>
+		);
+	}
+
+	const onClick = () => onSave( post, { ...edits, status: status || 'draft' }, blocks );
 
 	return (
-		<div className={ classes }>
-			<Dashicon icon={ icon } />
-			{ text }
-		</div>
+		<Button className={ classnames( className, 'button-link' ) } onClick={ onClick }>
+			{ wp.i18n.__( 'Save' ) }
+		</Button>
 	);
 }
 
 export default connect(
 	( state ) => ( {
+		post: getCurrentPost( state ),
+		edits: getPostEdits( state ),
+		blocks: getBlocks( state ),
 		isNew: isEditedPostNew( state ),
 		isDirty: isEditedPostDirty( state ),
+		isSaving: isSavingPost( state ),
+		status: getEditedPostStatus( state ),
+	} ),
+	( dispatch ) => ( {
+		onSave( post, edits, blocks ) {
+			dispatch( savePost( post.id, {
+				content: wp.blocks.serialize( blocks ),
```

Comment:
I wonder if block serialization should occur as part of a middleware before save proceeds into the network request handler, instead of the responsibility of the save button. What do you think?